test-runner/test_module_send_telemetry.py

```diff
@@ -40,3 +42,37 @@ def test_module_send_event_to_iothub():
     module_client.disconnect()
     log_message("disconnecting eventhub client")
     eventhub_client.disconnect()
+
+
+@pytest.mark.asyncio
+@pytest.mark.testgroup_edgehub_module_client
+@pytest.mark.testgroup_iothub_module_client
+@pytest.mark.callsSendEvent
+async def test_module_send_event_to_iothub_async():
+
+    executor = concurrent.futures.ThreadPoolExecutor()
```

Comment:
>executor = concurrent.futures.ThreadPoolExecutor() [](start = 4, length = 50)

Any particular reason we're using a custom ThreadPoolExecutor rather than using the event loop's default one?